cube/cube.cpp

```diff
@@ -703,69 +610,58 @@ void Demo::cleanup() {
     dfb->Release(dfb);
 #endif
 
-    inst.destroy(nullptr);
+    inst.destroy();
 }
 
 void Demo::create_device() {
-    float const priorities[1] = {0.0};
-
-    vk::DeviceQueueCreateInfo queues[2];
-    queues[0].setQueueFamilyIndex(graphics_queue_family_index);
-    queues[0].setQueueCount(1);
-    queues[0].setPQueuePriorities(priorities);
-
-    auto deviceInfo = vk::DeviceCreateInfo()
-                          .setQueueCreateInfoCount(1)
-                          .setPQueueCreateInfos(queues)
-                          .setEnabledLayerCount(0)
-                          .setPpEnabledLayerNames(nullptr)
-                          .setEnabledExtensionCount(enabled_extension_count)
-                          .setPpEnabledExtensionNames((const char *const *)extension_names)
-                          .setPEnabledFeatures(nullptr);
+    float priorities = {0.0};
+
+    std::vector<vk::DeviceQueueCreateInfo> queues;
+    queues.push_back(vk::DeviceQueueCreateInfo().setQueueFamilyIndex(graphics_queue_family_index).setQueuePriorities(priorities));
```

Comment:
The original calls `setPQueuePriorities()`, while the new calls `setQueuePriorities()`.  I think this is okay, but would prefer to draw your attention to it to double-check.

Reply:
Very intentional. The `setQueuePriorities()` function takes an array proxy, meaning it'll create a 'view' on the float object for us, rather than having to get a pointer to the float directly. more 'modern C++' (and more in line with the vulkan.hpp API)